### PR TITLE
[Merged by Bors] - chore(Topology/SubsetProperties): Refactor SubsetProperties.lean

### PR DESCRIPTION
**Pull request by loefflerd**
Split up the 2000-line `Topology/SubsetProperties.lean` into several smaller files. Not only is it too huge, but the name is very unhelpful, since actually about 90% of the file is about compactness; I've moved this material into various files inside a new subdirectory `Topology/Compactness/`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by grunweg** [comment]
Thank you for doing this! Just as a heads-up: this conflicts with #7576 (and also #7591).

**Comment by alexjbest** [comment]
@loefflerd was the reviewdog misbehaving here? Or being not as helpful as it could be somehow, I'd like to improve it if so but the history is now very confusing

File: Mathlib/Topology/Clopen.lean [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+import Mathlib.Topology.Bases
+import Mathlib.Data.Finset.Order
+import Mathlib.Data.Set.Accumulate
+import Mathlib.Data.Set.BoolIndicator
+import Mathlib.Topology.Bornology.Basic
+import Mathlib.Topology.LocallyFinite
+import Mathlib.Order.Minimal
+
+#align_import topology.subset_properties from "leanprover-community/mathlib"@"3efd324a3a31eaa40c9d5bfc669c4fafee5f9423"
+
```
Comment (ocfnash) [inline Mathlib/Topology/Clopen.lean:16]:
```suggestion
```

**Review by ocfnash — APPROVED** [review]
I have carefully checked over this (admittedly by hand) and I can attest that the following summary is accurate:

1. The file `Mathlib.Topology.SubsetProperties` has been renamed to `Mathlib.Topology.Compactness.Compact` as well as having its length cut from about 2k lines to about 1k lines.
2. The c.1k lines removed from `Mathlib.Topology.SubsetProperties` exactly correspond to the content added in the four new files `Topology.Clopen`, `Topology.Irreducible`, `Topology.Compactness.LocallyCompact`, `Topology.Compactness.SigmaCompact`. Indeed there is a character-by-character match except for file headers and four locations (occuring in the new file `Topology.irreducible`) where `α : Type u` was changed to `α : Type*`.
3. The file `Topology.Paracompact` was renamed to `Topology.Compactness.Paracompact`
4. Two results (`isClosed_and_discrete_iff` and `Filter.codiscrete`) were moved out of `Mathlib.Topology.SubsetProperties` and placed in `Topology.DiscreteSubset`.
5. File headers were added / updated according to the changes, in some cases by migrating comments from `Mathlib.Topology.SubsetProperties` (e.g., remarks about `IsPreirreducible`).
6. All other files changed have only had their `import` statements fixed to handle the new file layout.

Thanks for doing this.

bors d+

**Comment by PatrickMassot** [comment]
Thanks a lot @loefflerd! This is very tedious work that is very useful.

**Pull request closed without merging**